src/python/pants/util/dirutil.py

```diff
@@ -100,6 +101,7 @@ def safe_mkdir_for_all(paths):
       created_dirs.add(dir_to_make)
 
 
+@deprecated('1.16.0.dev2', hint_message='Use safe_file_write() instead!')
```

Comment:
Can we document the API changes between this and `safe_file_dump` please? E.g. 

```
Use safe_file_write() instead. It removes the deprecated `binary_mode` argument to instead only allow `mode`. It also defaults to writing unicode, rather than defaulting to bytes.
```

Reply:
Done!